Adding v1.1 of my USD example assets

RY-Storm opened the pull request:
As requested by Jeremy Cowles on the ASWF Slack, here are my USD example assets.

jcowles commented:
This is awesome, thank you so much!

Two initial notes:

1. I noticed one of the screenshots was over 1MB, is it possible to reduce the size?
2. Can the asset be renamed to reflect the contents of the asset (e.g. VehicleVariants etc).

(I haven't looked at the actual USD structure yet)

RY-Storm commented:
I am not sure why the Blender screenshot is the only one that is way bigger than the other screenshots, but I've lowered its quality now so that it is 458kb instead.

On the asset name side: I think all the USD assets are named clearly and accurately, such as 'wheelLargeRimAsset.usda', 'ambulanceBodyAsset.usda', etc. Do you mean those, and if so, what would you suggest to change them to?

If not, do you mean 'RYStorm-USD_Content_v1-1'? My thought process was to keep the name relatively bland so that anything can be added to it later. However, if you mean that this too unclear, then I have renamed it to 'Mini-Car-Examples'. This does mean that when in the future I add more USD content to this example, such as environments, characters, etc, I will also have to rename the folder accordingly to be representative again.

Renaming a folder can have disastrous USD consequences in case someone decides to reference this content, so I would like to keep it bland for that reason, but let me know what you think.

jcowles commented:
Oh yeah, I meant the top-level asset. In this repository, we don't expect people to have long lasting references to these files, they are all examples / educational / etc, so renaming should be fine.

Another way we could do this is make each of these an individual asset (Truck, Tractor, etc).

But if keeping them together is easier for you to add onto them, then I would just keep them together and evolve them as you see fit, without worrying about breaking stuff.

I think "Cars" or "Vehicles", I suggest this only because everything in this repo is intended to be a light weight example (so mini and example are always implied).

jcowles commented:
BTW, is it intentional to have the Truck geo in the Tractor?
<img width="283" alt="Screenshot 2023-09-02 at 11 37 11 AM" src="https://github.com/usd-wg/assets/assets/1510114/555e3989-8107-4f7a-9c6d-a63a6538ca13">

jcowles commented:
Would also be nice to add `defaultPrim` metadata to all the top level asset files and consider using payloads at the top level too.

jcowles commented:
Actually, if you're feeling inspired, maybe the most important thing you could do is explain the USD structure in the read me and your motivation for laying things out the way you did.

I think that would be helpful for anyone looking at this asset and would be aligned with your goal of making it easier for people to grok USD.

jcowles commented:
We have this doc: 
https://github.com/usd-wg/assets/blob/main/docs/asset-structure-guidelines.md

But the intent isn't for people to dogmatically follow it, rather we expect people to just use it as input to making their own decisions when building a pipeline - so explaining your choices would be a nice compliment to that information.

RY-Storm commented:
Renamed the main folder to 'Vehicles' and removed the extraneous asset, thank you.

For your other questions:
1. defaultPrim: Sure, happy to do that, but I will not be able to do that soon. Would be mid November at the earliest.
2. Payloads: Deliberately avoided to make sure the asset is simplified and easy to understand, which is the original intention of the asset pack as described here: https://www.youtube.com/watch?v=JixC53cQn5U
3. Explainer of USD structure: Intentionally kept simplified so that it is easier to flip through to see how it works, and is shown in the video above. My intention is to make a larger writeup and video to compare between various USD setups that are technically all valid, but have various pros & cons, including with environments.

jcowles commented:
Fair enough! Do you want to link to your video & blog posts from the readme instead?

jcowles commented:
I think you could give a super short preamble (like the explanation you gave in the USD WG sync) and then link to your problems with USD post, the asset post, and the video.

What do you think of that?

RY-Storm commented:
Sounds good. I've edited the readme and included the links in there. Does that work?

jcowles commented on full_assets/Vehicles/.~lock.readme.md#:
```diff
@@ -0,0 +1 @@
+,STORM/robin,Storm,02.09.2023 21:36,file:///C:/Users/robin/AppData/Roaming/LibreOffice/4;
```
Intentional?

RY-Storm replied:
Definitely not. Removed

jcowles commented:
Love the new readme! I noticed it looks like your last push picked up a random lock file, but other than that, I think it's good to go :)

jcowles commented:
Running usdchecker now (here in GitHub, see below)

RY-Storm commented:
I do not get any errors on my side. All assets also display fine, with their textures, in usdview and the DCCs mentioned in the readme. 

However, that log also says the checker would fail without default prims, so it sounds like closing this PR may take longer anyway. 

In this case I am guessing the checker wants a _particular_ USD setup, not necessarily a _required_ USD setup. I will see if I have time tomorrow to look into what may be causing this, but I would also like to know what the checker is requiring, and why.

RY-Storm commented:
If the above is indeed the problem, then I can find some good humor in that the assets made for the benefit of the article are being halted due to exact issue the article mentions at heading 5.

jcowles commented:
I'll take a quick look now and see if I can track down the source of errors

jcowles commented:
I think you're right, something is funky about the way we're running usdchecker - I think we should merge this without passing, but use this asset to discuss usdchecker and why this is happening

pablode commented:
usdchecker is right with the texture references; they're defined using backslashes ('\\'), which is not portable.

RY-Storm commented:
@pablode I looked through the files, both the content and materials, and found 0 uses of a backslash ('\').

When I go to the 'Files Changed' tab here on github and search for \ I also get 0 hits, while / gives 1241 hits.

Am I misunderstanding, is there a miscommunication, or does usdchecker have an issue?

pablode commented:
```
--- full_assets/Vehicles » grep -rIF '\' .
./USD_Mini_Car_Kit/assets/wheels/wheelRed/geo/wheelRedGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelRed/geo/wheelRedGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\red.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelRed/geo/wheelRedGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelNormal/geo/wheelNormalGeo.usda:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelNormal/geo/wheelNormalGeo.usda:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelLargeRim/geo/wheelLargeRimGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelLargeRim/geo/wheelLargeRimGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelWide/geo/wheelWideGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelWide/geo/wheelWideGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelVintage/geo/wheelVintageGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\white.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelVintage/geo/wheelVintageGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelVintage/geo/wheelVintageGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelBlack/geo/wheelBlackGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/wheels/wheelBlack/geo/wheelBlackGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\backlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\red.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\frontlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/tractor/geo/tractorGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\blue.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-textures\frontlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-textures\backlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/van/geo/vanGeo.usd:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\white.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\red.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\frontlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\blue.jpg@
./USD_Mini_Car_Kit/assets/vehicles/ambulance/geo/ambulanceGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/formula/geo/formulaGeo.usda:        asset inputs:file = @..\..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/vehicles/formula/geo/formulaGeo.usda:        asset inputs:file = @..\..\..\..\textures\global-colors\red.jpg@
./USD_Mini_Car_Kit/assets/vehicles/formula/geo/formulaGeo.usda:        asset inputs:file = @..\..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/formula/geo/formulaGeo.usda:        asset inputs:file = @..\..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greymedium.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\green.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-colors\greylight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\frontlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/4wd/geo/4wdGeo.usd:        asset inputs:file = @..\..\..\..\textures\global-textures\backlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/sedan/geo/sedanGeo.usda:        asset inputs:file = @..\..\..\textures\global-textures\frontlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/sedan/geo/sedanGeo.usda:        asset inputs:file = @..\..\..\textures\global-textures\backlight.jpg@
./USD_Mini_Car_Kit/assets/vehicles/sedan/geo/sedanGeo.usda:        asset inputs:file = @..\..\..\textures\global-colors\red.jpg@
./USD_Mini_Car_Kit/assets/vehicles/sedan/geo/sedanGeo.usda:        asset inputs:file = @..\..\..\textures\global-textures\window.jpg@
./USD_Mini_Car_Kit/assets/vehicles/sedan/geo/sedanGeo.usda:        asset inputs:file = @..\..\..\textures\global-colors\greylight.jpg@
```
https://github.com/usd-wg/assets/pull/53/files#diff-22ffdf8078a48af012c1d67a523388e4e9ea02720b63f88467f2ba4630c759bbR126

RY-Storm commented:
Thank you! I've updated the files, do these work now?

Also, I have tried to puzzle through what happened here:

1. 3dsMax uses \ for directory references by default. It does not accept /.
2. I have to manually re-write the USD files after export from 3dsMax so they have / instead of \. 3dsMax's USD exporter does not automatically fix this.
3. Houdini does by default add a /, so the assets from there are fine.
4. GitHub does not by default show diffs of large files, so I could not find the \ by searching on the diff page. I had to open all files in Sublime and do a search from there instead to catch it.

Sounds like there is potentially a point of action here, to let Autodesk know that there may be a need for 3dsMax to use / by default USD files, or have their USD exporter automatically change \ to / on export.

RY-Storm commented:
I've now also added default prims to all Houdini exported files. In this case 3dsMax's USD exporter does add default prims by default, so I thankfully did not have to manually fix anything there.

The Houdini exported USDs did require some fixes on the nodes to set default prims, but that _should_ work now.

jcowles commented:
Looks like there still may be an issue with the texture path in usdchecker, notice the path to the textures is `.../USD_Mini_Car_Kit/assets/textures/...`, but it should be `.../USD_Mini_Car_Kit/textures/...` (no "assets").

And thank you so much for iterating on this to get it to the finishline, I know this is probably more than you thought you were signing up for, but it's very much appreciated!

RY-Storm commented:
For posterity: We are looking into why usdchecker gets this issue by chatting about it in the ASWF Slack. The assets themselves _seem_ fine so far. Or at least: We cannot find an issue in the usda's.

RY-Storm commented:
Removed all the obsolete material references from the geo files, which were added at the time through the export from 3dsMax. Maybe this resolves the issue?

RY-Storm commented:
Failed :(
```
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_1_greyMediumMax>. (fails 'MaterialBindingAPIAppliedChecker')
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_2_greenMax>. (fails 'MaterialBindingAPIAppliedChecker')
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_3_windowMax>. (fails 'MaterialBindingAPIAppliedChecker')
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_5_greyLightMax>. (fails 'MaterialBindingAPIAppliedChecker')
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_6_frontLightMax>. (fails 'MaterialBindingAPIAppliedChecker')
Found material bindings but no MaterialBindingAPI applied on the prim </_wd/_7_backLightMax>. (fails 'MaterialBindingAPIAppliedChecker')
```

I will not be able to look at this for the next few weeks. I'll have time again at the end of October. Please do let me know if you find any issues here though, or in Slack.

pablode commented:
Almost looks fine! https://github.com/pablode/usd-wg-assets/actions/runs/6124318276/job/16624192769
For the MaterialBindingAPI errors, you can run `usdfixbrokenpixarschemas` on the USD files.

jcowles commented:
Looks like it's just missing the MaterialBindingAPI applications

beersandrew commented:
Hope this isn't rude or overstepping, but I have a fix for this a posted to @RY-Storm, looks like he's away for a bit and I learned that I could post it directly here as another PR: https://github.com/usd-wg/assets/pull/58

meshula commented:
I think it would be okay to land Andrew's patch over this one at #58, and do more iteration as necessary?
